os/board/rtl8721csm/src/component/os/tizenrt/lwip_intf_tizenrt.c

```diff
@@ -231,11 +232,13 @@ unsigned char *netif_get_hwaddr(int idx_wlan)
 {
 	struct netdev *dev_tmp = NULL;
 	unsigned char *dev_addr;
+	dev_addr = (unsigned char *)malloc(sizeof(unsigned char) * (IFHWADDRLEN));
 
 	dev_tmp = rtk_get_netdev(idx_wlan);
 
 	if (!dev_tmp) {
 		DiagPrintf("[netif_get_hwaddr] get dev fail\n");
+		return NULL;
```

Comment:
For memory allocation from line 235 which you added, is it okay not to free the memory for failure case?
If needed, please follow the comment which I added above.
I mean, check the memory usage and use `free` or `kmm_free`.

Reply:
Thanks a lot! You are right. 
This function actually is used nowhere, so if we malloc inside this function, we do not have the successful case yet and thus cannot free. How about I change the function to bellow? Then for future usage, users can handle malloc/free properly.

int netif_get_hwaddr(int idx_wlan, uint8_t *dev_addr)
{
	struct netdev *dev_tmp = NULL;

	dev_tmp = rtk_get_netdev(idx_wlan);

	if (!dev_tmp) {
		DiagPrintf("[netif_get_hwaddr] get dev fail\n");
		return -1;
	}
	if (netdev_get_hwaddr(dev_tmp, dev_addr, (unsigned char*)IFHWADDRLEN) == 0)
		return 0;
	else
		return -1;
}

Reply:
Sorry, but I didn't understand well what you are saying.
Do you mean that you will not allocate the memory?
Or you want that user will free the memory if user get the `-1` when calling `netif_get_hwaddr()`?

Reply:
Hi, Mr Kim. I mean I add dev_addr to input of this API, then when users try to use this API, they can malloc the memory for it, and free in both success and failure cases. The API itself will not malloc memory for it instead.

Reply:
Sorry I missed this discussion and merged this PR already. Could you make a new PR for this?

Reply:
@fengyunlai I agree your change. 
```
int netif_get_hwaddr(int idx_wlan, uint8_t *dev_addr)
{
	struct netdev *dev_tmp = NULL;

	dev_tmp = rtk_get_netdev(idx_wlan);

	if (!dev_tmp) {
		DiagPrintf("[netif_get_hwaddr] get dev fail\n");
		return -1;
	}
	if (netdev_get_hwaddr(dev_tmp, dev_addr, (unsigned char*)IFHWADDRLEN) == 0)
		return 0;
	else
		return -1;
}
```

Reply:
Thank you Mr Chang!